src/transformers/trainer.py

```diff
@@ -3831,6 +3831,13 @@ def evaluation_loop(
             if is_torch_xla_available():
                 xm.mark_step()
 
+            if "use_gather_object" in inspect.signature(
+                self.gather_function
+            ).parameters.keys() and is_accelerate_available("0.30.0"):
+                self.gather_function = functools.partial(
+                    self.gather_function, use_gather_object=self.args.eval_use_gather_object
+                )
+
```

Comment:
In case the user don't have accelerate > 0.30.0 but set `use_gather_object = True`, could you return a warning to let the user know that he should update its version of accelerate ?

Reply:
@SunMarc https://github.com/huggingface/transformers/pull/31514/commits/f44dabf43a1d1ab244a1f1c87cf80c367563d072
How about now?